infrastructure/src/main/java/org/corfudb/infrastructure/management/ClusterRecommendationEngine.java

```diff
@@ -0,0 +1,56 @@
+package org.corfudb.infrastructure.management;
+
+import org.corfudb.protocols.wireprotocol.NodeView;
+import org.corfudb.protocols.wireprotocol.PeerView;
+import org.corfudb.runtime.view.Layout;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link ClusterRecommendationEngine} provides methods to decide about failure or healing
+ * status of Corfu servers in a given {@link Layout} and for a specific view of the cluster
+ * captured in a {@link NodeView}.
+ * Decisions are dependant on the concrete underlying algorithm corresponding to a
+ * {@link ClusterRecommendationPolicy}.
+ *
+ * Created by Sam Behnam on 10/19/18.
+ */
+public interface ClusterRecommendationEngine {
+
+    /**
+     * Get the corresponding {@link ClusterRecommendationPolicy} used in the current instance of
+     * {@link ClusterRecommendationEngine}. This policy represents the characteristics of the
+     * underlying algorithm used for making a decision about suspecting the failed or healed
+     * status of Corfu servers.
+     *
+     * @return a concrete instance of {@link ClusterRecommendationPolicy}
+     */
+    ClusterRecommendationPolicy getClusterRecommendationPolicy();
+
+    /**
+     * Provide a list of servers in the Corfu cluster which according to the underlying algorithm
+     * for {@link ClusterRecommendationPolicy} are suspected to have been failed. The decision
+     * is made based on the given view of the cluster captured in {@link NodeView} along with the
```

Comment:
NodeView here seems not consistent with the method signature.

Reply:
It was from the time that aggregation was happening to NodeView. Since the aggregation has changed to provide ClusterState and NodeState to represent the cluster, I'll be updating these to use them accordingly.